src/main/kotlin/espressoLogcat/EspressoLogcat.kt

```diff
@@ -1,348 +1,304 @@
 package espressoLogcat
 
+import espressoLogcat.Formatters.Companion.createLineDataMap
+import espressoLogcat.Formatters.Companion.formatLineData
+import espressoLogcat.Formatters.Companion.getUnmatchedLineData
 import espressoLogcat.OutputFormat.*
-import java.io.BufferedReader
-import java.io.InputStreamReader
 import java.time.format.DateTimeFormatter
+import kotlin.math.log
 
-class EspressoLogcat {
-    companion object {
-        fun getLogcatLikeBrief(
-            tag: String = "*",
-            priority: Priority? = null,
-            bufferedReaderSize: Int = 4 * 1024,
-            dateTimeFormatter: DateTimeFormatter = DateTimeFormatter.ISO_DATE_TIME,
-            regex: Regex? = null,
-            filename: String? = null,
-            tCount: Int? = null,
-            tTime: String? = null,
-            pid: Int? = null,
-            vararg options: String? = arrayOf()
-        ): MutableCollection<String> {
-            return EspressoLogcat().getLogCat(
-                tag = tag,
-                priority = priority,
-                bufferedReaderSize = bufferedReaderSize,
-                dateTimeFormatter = dateTimeFormatter,
-                regex = regex,
-                filename = filename,
-                tCount = tCount,
-                tTime = tTime,
-                pid = pid,
-                outputFormat = BRIEF,
-                options = options
-            )
-        }
+class EspressoLogcat(private val logcatExecutor: LogcatExecutor = LogcatExecutor()) {
```

Comment:
You'd want the argument type to be the interface so that you can pass in an arbitrary implementation of it. So like 

```class EspressoLogcat(private val logcatExecutor: ILogcatExecutor = LogcatExecutor())```

Reply:
Fixed in bb55ef4